### PR TITLE
feat: make generating of execution numbers atomic

### PR DESCRIPTION
**Pull request by nicufk**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — CHANGES_REQUESTED** [review]
we need to add selector to Webhook CRD

**Review by vsukhin — COMMENTED** [review]
one question

**Review by vsukhin — COMMENTED** [review]
.

File: internal/pkg/api/repository/result/mongo.go [hunk 1]
```diff
@@ -14,16 +14,26 @@ import (
 	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
 )
 
-const CollectionName = "results"
+const (
+	CollectionName    = "results"
+	CollectionNumbers = "numbers"
```
Comment (exu) [inline internal/pkg/api/repository/result/mongo.go:19]: I would name it 'sequences'

File: internal/pkg/api/repository/result/interface.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ type Repository interface {
 	GetExecutions(ctx context.Context, filter Filter) ([]testkube.Execution, error)
 	// GetExecutionTotals gets the statistics on number of executions using a filter, but without paging
 	GetExecutionTotals(ctx context.Context, paging bool, filter ...Filter) (result testkube.ExecutionsTotals, err error)
+	// GetNextExecutionNumber gets next execution number by test name
+	GetNextExecutionNumber(ctx context.Context, testName string) (number int, err error)
```
Comment (exu) [inline internal/pkg/api/repository/result/interface.go:46]: I would close it in separate interface and add it here 

Additionally move out logic to separate structure to allow easy composition

**Review by vsukhin — APPROVED** [review]

**Review by exu — APPROVED** [review]

**Pull request merged**